cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/upgrade/AwsRdsUpgradeService.java

```diff
@@ -40,8 +40,10 @@ public void upgrade(AuthenticatedContext ac, DatabaseStack dbStack, Version targ
         AmazonRdsClient rdsClient = getAmazonRdsClient(ac);
         RdsInfo rdsInfo = getRdsInfo(dbInstanceIdentifier, rdsClient);
         validateIfRdsCanBeUpgraded(targetMajorVersion, rdsInfo);
-        upgadeRdsIfNeeded(targetMajorVersion, databaseServer, rdsClient, rdsInfo);
-        waitForRdsUpgrade(ac, databaseServer, rdsClient);
+        if (awsRdsUpgradeValidatorService.isRdsMajorVersionSmallerThanTarget(rdsInfo, targetMajorVersion)) {
+            upgadeRdsIfNeeded(targetMajorVersion, databaseServer, rdsClient, rdsInfo);
```

Comment:
could you pls come up with a more descriptive name for this method? It is not clear what `IfNeeded` means in this context, it looks like it performs additional validations but it does not.